reve/reve/lib/ModuleSimplifier.cpp

```diff
@@ -355,6 +360,166 @@ llvm::PreservedAnalyses RemoveLifetimeCallsPass::run(
+                for(llvm::Use &UU : II->uses()) {
+                    llvm::errs() << "  "; UU.getUser()->print(llvm::errs(), false); llvm::errs() << "\n";
+                }
+#endif
+                if(!II->use_empty()) // The return value is actually used
+                    can_replace = false;
+            } else can_replace = false; // The function is used somewhere as an argument, therefore
+                                        // it ought not to be replaced
+        }
+        
+        if(can_replace) {  
+            // Create the header of the new function
+            std::vector<llvm::Type*> FAT_New (Fun.getFunctionType()->param_begin(), Fun.getFunctionType()->param_end());
+            llvm::FunctionType *FT_New = llvm::FunctionType::get(llvm::Type::getVoidTy(Fun.getContext()), FAT_New, Fun.isVarArg());
+            llvm::Function *Fun_New = llvm::Function::Create(FT_New, Fun.getLinkage(), Fun.getName(), Fun.getParent());
+            
+            // Copy the attributes from the old function and delete the ones related to the return value
+            Fun_New->copyAttributesFrom(&Fun);
+            for(llvm::Attribute::AttrKind AK : badAttributes) {
+                Fun_New->removeAttribute(llvm::AttributeList::ReturnIndex, AK);
+                Fun_New->removeAttribute(llvm::AttributeList::FunctionIndex, AK);
+            }
+            Fun_New->takeName(&Fun);
+            
+            for(llvm::Function::arg_iterator AI = Fun.arg_begin(), AE = Fun.arg_end(), NAI = Fun_New->arg_begin();
+                AI != AE; 
+                ++AI, ++NAI)
+            {
+                NAI->takeName(AI); // Set the names of all arguments of the new function
+            }
+            
+            Fun_New->getBasicBlockList().splice(Fun_New->begin(), Fun.getBasicBlockList()); // Copy the function body
+            
+            // Replace return instructions on ends of basic blocks with ret void
+            for(llvm::BasicBlock &B : *Fun_New)
+                if(llvm::dyn_cast<llvm::ReturnInst>(B.getTerminator())) {
+                    B.getInstList().pop_back();
+                    llvm::ReturnInst *Term_New = llvm::ReturnInst::Create(B.getContext());
+                    B.getInstList().push_back(Term_New);
+                }
+            
+            for(llvm::Function::arg_iterator I = Fun.arg_begin(), E = Fun.arg_end(), NI = Fun_New->arg_begin();
+                I != E;
+                ++I, ++NI)
+            {
+                I->replaceAllUsesWith(NI); // Replace all uses of the old arguments
+            }
+            
+            // For call or invoke instructions a new instruction has to be created and the old one replaced
+            for(llvm::Use &U : Fun.uses()) {
+                llvm::CallInst* CI = llvm::dyn_cast<llvm::CallInst>(U.getUser());
+                llvm::InvokeInst* II = llvm::dyn_cast<llvm::InvokeInst>(U.getUser());
+                
+                if(CI) {
```

Comment:
Use:
if (llvm::CallInst* CI = llvm::dyn_cast<llvm::CallInst>(U.getUser())) {
    ....
}
Same applies to II.